src/main/java/controller/LevelController.java

```diff
@@ -208,6 +216,41 @@ public void handle(long now) {
     }
 
     /**
+     * This is the cycle that performs all powerups operations.
+     * @param powerup The powerup actions are performed on
+     * @param player The player there might be a collision with.
+     */
+    public void performPowerupsCycle(Powerup powerup, Player player) {
+        powerup.causesCollision(player);
+        powerup.move();
+    }
+
+    /**
+     * This is the cycle that performs all player operations.
+     * @param player The player the actions are performed on.
+     */
+    private void performPlayerCycle(Player player) {
+        player.processInput();
+        player.move();
+        player.checkBubbles();
+        player.getBubbles().forEach(Bubble::move);
+    }
+
+    /**
+     * This function updates the powerups, and removes
+     * the ones which have been picked up.
+     */
+    public void updatePowerups() {
+        ArrayList<Powerup> nPowerups = new ArrayList<>();
+        for (Powerup powerup : powerups) {
```

Comment:
I think you shouldn't worry about the naming here, since it is a default styling for looping through a list. We also do this with Players, Walls and Monsters.

<naming